### PR TITLE
Update android example

### PR DESCRIPTION
**Pull request by mkovalyk**
*Issue #, if available:#[601](https://github.com/amzn/style-dictionary/issues/601)

*Description of changes:*
A lot of changes have been made to ensure sample is up-to-date:

- Convert to Kotlin. Version - 1.4.32
- Remove Butterknife dependency
- Convert to androidx library
- Upgrade Jackson library to 2.12.3
- Update gradle to 6.8.2 Gradle plugin 4.1.3 
- Compile and Target SDK - 30

Things left as they were: 
 - No reaction on Sizes button 
image
 - Crash on clicking for non-color resource on "All Properties" screen:
image

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/complete/android/demo/src/main/java/com/amazon/styledictionaryexample/color/BackgroundColorActivity.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+      holder.item = mValues[position]
+        .apply {
+          val id = resources.getIdentifier(name, "color", packageName)
+          holder.swatchView.setBackgroundColor(resources.getColor(id, null))
+          holder.titleView.text = attributes["item"]
+          holder.view.setOnClickListener {
+            //                    Context context = v.getContext();
```
Comment (dbanksdesign) [inline examples/complete/android/demo/src/main/java/com/amazon/styledictionaryexample/color/BackgroundColorActivity.kt:46]: Should we remove these commented out lines?

File: examples/complete/android/demo/src/main/java/com/amazon/styledictionaryexample/color/BaseColorActivity.kt [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+        if (holder.valueView != null) {
+          holder.valueView.text = subtitle
+          holder.valueView.setTextColor(fontColor)
+        }
+
+        holder.view.setBackgroundColor(resources.getColor(id, null))
+        holder.view.setOnClickListener {
+        }
+        //                    Context context = v.getContext();
```
Comment (dbanksdesign) [inline examples/complete/android/demo/src/main/java/com/amazon/styledictionaryexample/color/BaseColorActivity.kt:115]: Remove these lines? Or it looks like there is a ColorDetailFragment that maybe this was intended to navigate to?

File: examples/complete/android/demo/src/main/java/com/amazon/styledictionaryexample/models/Property.kt [hunk 3]
```diff
@@ -1,35 +1,28 @@
 /**
  * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
  *
+ *
```
Comment (dbanksdesign) [inline examples/complete/android/demo/src/main/java/com/amazon/styledictionaryexample/models/Property.kt:4]: I think these additional comment lines got added on accident?

File: examples/complete/android/demo/src/main/java/com/amazon/styledictionaryexample/models/Property.kt [hunk 4]
```diff
@@ -1,35 +1,28 @@
...
-    public String name;
-    public String value;
-    public Map<String, String> attributes;
-    public ArrayList<String> path;
-
-    public Property() {}
-}
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
 
+@JsonIgnoreProperties("original", "filePath", "isSource")
```
Comment (dbanksdesign) [inline examples/complete/android/demo/src/main/java/com/amazon/styledictionaryexample/models/Property.kt:22]: 🙌

File: examples/complete/android/demo/src/main/java/com/amazon/styledictionaryexample/models/StyleDictionaryNode.kt [hunk 5]
```diff
@@ -1,23 +1,24 @@
+package com.amazon.styledictionaryexample.models
```
Comment (dbanksdesign) [inline examples/complete/android/demo/src/main/java/com/amazon/styledictionaryexample/models/StyleDictionaryNode.kt:1]: I think this should be moved below the license comment

File: examples/complete/android/demo/src/main/java/com/amazon/styledictionaryexample/util/StringHelper.kt [hunk 6]
```diff
@@ -1,22 +1,21 @@
-package com.amazon.styledictionaryexample.models;
-
 /**
  * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
- * <p>
```
Comment (dbanksdesign) [inline examples/complete/android/demo/src/main/java/com/amazon/styledictionaryexample/util/StringHelper.kt:5]: 🙌

**Review by dbanksdesign — COMMENTED** [review]
Thank you so much for this great contribution! This code has already helped one person get up and running on Android where the current example is broken. I just had a few very minor comments, but overall this looks great!

**Comment by chazzmoney** [comment]
@mkovalyk Would love to merge this in and have it as part of our upcoming 3.0 release.

Let us know if you need any support in making the minor changes remaining.

Thanks!

File: examples/complete/android/demo/src/main/java/com/amazon/styledictionaryexample/color/BaseColorActivity.kt (hunk 2 above)
Comment (mkovalyk) [reply examples/complete/android/demo/src/main/java/com/amazon/styledictionaryexample/color/BaseColorActivity.kt:115]: These lines were commented out in original files. Didn't want to change too much or add the logic which was not added originally. I can make it in the scope of separate PR. Thanks for your comment!

File: examples/complete/android/demo/src/main/java/com/amazon/styledictionaryexample/models/Property.kt (hunk 3 above)
Comment (mkovalyk) [reply examples/complete/android/demo/src/main/java/com/amazon/styledictionaryexample/models/Property.kt:4]: Yeah! Looks like after automatic conversion to Kotlin. Removed

**Comment by mkovalyk** [comment]
@chazzmoney @dbanksdesign Thanks for the comments! I have resolved all of them so hopefully, there are no more blockers for this PR. I have rebased the branch on top of the latest 3.0 branch. Not sure if it aligns with the project's branching policy, but I couldn't find any warning not to do so :)

**Review by chazzmoney — APPROVED** [review]
LGTM :shipit:

**Pull request merged**